### PR TITLE
Remove legacy code in blockchain.cpp

### PR DESCRIPTION
**Pull request by LefterisJP**

### CODE REVIEW

**Comment by gavofyork** [comment]
feel free to move `contains` to somewhere else, but i find it infinitely more readable than the cryptic and verbose STL alternative. man i hate the STL.

**Comment by LefterisJP** [comment]
But it's a function inside a .cpp file that's used in only one place which does exactly what an STL function does. I will really have to insist on this PR. I don't believe that replicating STL or not should be a matter of personal preference. It should just not happen.

**Comment by gavofyork** [comment]
> feel free to move contains to somewhere else

like a .h file.

**Comment by LefterisJP** [comment]
I still believe it to be unecessary but I am willing to compromise.

Rebased on develop and moved it to CommonData.h

File: libdevcore/CommonData.h [hunk 1]
```diff
@@ -334,4 +334,10 @@ std::vector<T> keysOf(std::unordered_map<T, U> const& _m)
 	return ret;
 }
 
+template <class T, class V>
+bool contains(T const& _t, V const& _v)
+{
+	return std::end(_t) != std::find(std::begin(_t), std::end(_t), _v);
```
Comment (gavofyork) [inline libdevcore/CommonData.h]: std::end could conceivably be non-trivial. so i'd prefer:

```c++
auto e = std::end(_t);
return e != std::find(std::begin(_t), e, _v);
```

**Comment by LefterisJP** [comment]
I am pretty sure that all of the compilers we use would perform Common Subexpression Elimination on this and optimize the extra `std::end()` away.

**Comment by subtly** [comment]
@LefterisJP On to getting contains() into the next C++ standard? :+1:

**Comment by gavofyork** [comment]
they'd still insist on having to pass three args in (begin, end, value) rather than two (container, value) like Every. Other. Language.

**Pull request merged**